mobile/src/features/Analytics/hooks/usePageViewTracking.ts

```diff
@@ -0,0 +1,49 @@
+import {useFocusEffect} from '@react-navigation/native'
+import * as React from 'react'
+
+import type {
+  AnalyticsEvent,
+  AnalyticsEventProperties,
+} from '../events/analytics-events'
+import {useAnalyticsTracking} from './useAnalyticsTracking'
+
+/**
+ * Hook to automatically track page views when a screen comes into focus
+ */
+export function usePageViewTracking<T extends AnalyticsEvent>(
+  event: T,
+  properties?: T extends keyof AnalyticsEventProperties
+    ? AnalyticsEventProperties[T]
+    : never,
+) {
+  const {trackEvent} = useAnalyticsTracking()
+
+  useFocusEffect(
+    React.useCallback(() => {
+      trackEvent(event, properties)
+    }, [trackEvent, event, properties]),
+  )
+}
+
+/**
+ * Hook for tracking page views with a delay (useful for screens with loading states)
+ */
+export function useDelayedPageViewTracking<T extends AnalyticsEvent>(
```

Comment:
Seems redundant to have 2 hooks doing the same, could just add the optional delay to the main one.

Reply:
good point. thanks